subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -611,7 +611,7 @@ public Node selectNext(WorkerLeaseRegistry.WorkerLease workerLease, ResourceLock
                     node.startExecution();
                 } else {
                     node.skipExecution();
-                    updateAllDependenciesCompleteForPredecessors(node);
+                    recordNodeCompleted(node);
```

Comment:
I wonder if we could do something like `Node.completeNode(Consumer<Node>)` so we could do this:

```java
completeNode(node, Node::skipExecution);
```

What I'm trying to achieve here is to remove the risk of calling `skipExecution()` etc. without completing the node.

Same with `recordNodeStarted()`, though that's only called from one place, so it's more about consistency. Another solution would be to rename `recordNodeStarted()` to `startNode()` and let it call `node.startExecution()`.

Reply:
`aborted` could also be set by `completeNode()` explicitly.

Reply:
The completion and starting methods in `Node` now take an extra consumer for the bookkeeping.